JWT/JWTTools.swift

```diff
@@ -0,0 +1,301 @@
+//
+//  JWTTools.swift
+//  uPortSDK
+//
+//  Created by Cornelis van der Bent on 20/12/2018.
+//
+
+import Foundation
+import BigInt
+import CoreEthereum
+import Security
+
+public enum JWTToolsError: Error
+{
+    case notValidIssuedInFuture
+    case notValidPastExpiryDate
+    case malformedNotThreeParts
+    case malformedEmptyHeader
+    case malformedEmptyPayload
+    case malformedEmptySignature
+    case malformedNotBase64Url
+    case malformedNotBase64
+    case malformedNotUtf8
+    case malformedNotDictionary
+    case deserializationError(String)
+    case invalidSignatureSize(Int)
+    case failedToVerify
+    case missingDidDocument
+}
+
+/// Protocol to support testing.
+public protocol JWTToolsDateProvider
+{
+    func now() -> Date
+}
+
+/// Class with JWT related functionality.
+public struct JWTTools
+{
+    private struct Constants
+    {
+        static let signatureSize = 64
+        static let timeSkew = TimeInterval(0.3)
```

Comment:
this skew should be 5 minutes.

Reply:
I misread the Android code and already found 300ms extremely short. Glad I was 'wrong'.